ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/EthHashBlockCreator.java

```diff
@@ -104,8 +104,8 @@ private EthHashSolverInputs generateNonceSolverInputs(
     return nonceSolver.hashesPerSecond();
   }
 
-  public boolean submitWork(final EthHashSolution solution) {
```

Comment:
Do we have to go to just a long here?  Can EthHashSoluton remain?

Reply:
In the case of eth_submitWork, you can compute a EthHashSolution object from the JSON-RPC request. In the case of stratum, that's not as easy, but you can, if you keep the EthHashInputs object around.

However, that work is duplicated, since this method discards the solution object, and instead recomputes one in the testNonce method. It also has the advantage of keeping track of the inputs in just one place that can be the up-to-date reference for the current job.

Reply:
OK, I tried again with ethminer, with http and stratum+tcp, and managed to make sure we could use the parameters passed back to form a solution. So this is squared away now.